### PR TITLE
gnuradio-osmosdr: 0.1.4 -> 4d83c60, Added support for Soapysdr

### PR DESCRIPTION
**Pull request by betaboon**
###### Motivation for this change
I needed support for limeSDR in gnuradio and gqrx,
This can be accomplished by using limesuite via soapysdr and soapysdr via gr-osmosdr.

So this results in a rebuild of:
- `gnuradio-companion`
- `gqrx`
- `qradiolink`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by markuskowa** [comment]
Does gqrx work for you with the soapy drivers? I did try several times over last year and could never get it to work.

**Comment by markuskowa** [comment]
@bjornfor @the-kenny 
How do you feel about upgrading from a stable version to a development version? The argument for upgrading would be that the latest stable is from 2014 and many new features have been implemented since then.

**Comment by betaboon** [comment]
@markuskowa 
limesdr-gqrx-nixos

**Comment by bjornfor** [comment]
@markuskowa: My personal opinion is that nixpkgs should only package released / stable upstream software. My thinking is that if a project, like osmosdr, has not produced a release in a long time, and the thinking is that users would benefit from getting the latest code, we should strive for making a release instead of working around it in nixpkgs. Think of other downstreams, if upstreams never release, all downstreams would eventually have to consider moving to unstable/unreleased code. And I think very few projects have a development process where _all_ commits to git master branch are stable enough to be safely distributed to a wider user base.

> The argument for upgrading would be that the latest stable is from 2014 and many new features have been implemented since then.

Then I would say that's a good argument for upstream to make a release :-)

**Comment by markuskowa** [comment]
@bjornfor I wrote an email to one of the developers a year ago, asking about a release, but did not get response. There is certainly a good argument to make a release, specially since the development branch seems to be pretty stable (works fine for me locally).

**Comment by betaboon** [comment]
going through the mailing-list archive i found:
- http://lists.osmocom.org/pipermail/osmocom-sdr/2018-October/001847.html
- http://lists.osmocom.org/pipermail/osmocom-sdr/2018-June/001766.html

I just hopped into irc://freenode#osmocom just asking around

**Comment by bjornfor** [comment]
> @bjornfor debian uses ~90 patches to compensate for the old version

Wow.

> I understand the argument that we do not want to switch to unstable. Should we keep this PR open as "waiting-for-upstream" and hope that there will be a release soon or is there maybe some other middle ground solution?

I don't know what middle ground would be. I see these options:

0. Kindly ask upstream again to make a release. Offer help.
1. Do a local distro-specific fork of the latest release (apparently how debian does it for this package).
2. Do a real fork the project, since upstreams that don't produce releases are effectively dead (IMHO).
3. Switch to unstable and update to arbitrary commits.

Since I'm not an active maintainer (just a guy that played with SDR some time ago) I'm not going to say hard no to option 3. (But you know I prefer option 0.)

**Comment by markuskowa** [comment]
Let's see if we get a response from the maintainers in the near future.

**Comment by greydot** [comment]
Could you also add libbladeRF dependency so that there would be no need for a separate PR?
Thank you.

**Comment by greydot** [comment]
@markuskowa it seems that other distros just use an unstable version.

**Comment by markuskowa** [comment]
@greydot Yes, Debian uses ~90 patches but still calls it v0.1.4. It is essentially an unstable version.
Does the latest bladerf support require osmosdr unstable? (there are a lot of bladerf related commits on the current master of osmosdr)

**Comment by greydot** [comment]
@markuskowa yes. v1.1.4 doesn't build with the latest libbladerf. The unstable code works just fine. See my closed PR linked above.

**Comment by Mic92** [comment]
I'd say unstable should be preferred in that case. Releases are worthless if not happening regularly.

**Comment by markuskowa** [comment]
OK, it looks like the consensus is to go for the unstable version. We still have some to time to work out potential problems until 19.03. 
@betaboon I wait for your reply to the libbladerf request.

**Comment by betaboon** [comment]
I talked to someone called `Hoernchen` in irc://freenode#osmocom who is a longtime contributor to gr-osmosdr. he agreed that a release should be created (after i mentioned the debian-90+patch situation).
the gr-osmosdr maintainer called `horizon`(on irc) is thought to be quite active in irc, but went offline a couple of days ago and might be on vacation. i will keep an eye on him coming back online and keep pushing for a proper release.

For the time being I would be in favour of using a pinned commit in `master`.

@greydot @markuskowa regarding the libbladeRF support. i have no way of testing that it works as i dont own a bladerf.
currently `soapybladedrf` is included in `soapysdr-with-plugins` which is being used here. so bladerf might already work going the `libbladerf`->`soapybladerf`->`soapysdr`->`gr-osmosdr` route.

If that is not sufficient for you (or you prefer `libbladerf`->`gr-osmosdr`) i could add the dependency and test up to the point of having it compile. but as stated: i cannot test with a bladerf.

**Comment by markuskowa** [comment]
@betaboon I guess it is fine to add `libbladerf`. @greydot can test it (or already did test it in the other PR)?

**Comment by greydot** [comment]
@markuskowa I did test it successfully in the other PR with gqrx and osmocom_fft.

**Comment by greydot** [comment]
When can we expect this (plus bladerf support) merged?

File: pkgs/applications/misc/gnuradio/osmosdr.nix
Comment (markuskowa) [inline pkgs/applications/misc/gnuradio/osmosdr.nix]: `soapysdr-with-plugins` needs to be conditional for linux otherwise it won't build on darwin anymore.

File: pkgs/applications/misc/gnuradio/osmosdr.nix
Comment (markuskowa) [inline pkgs/applications/misc/gnuradio/osmosdr.nix]: Can we change that to dashes?
```suggestion
  version = "2018-08-15";
```

**Comment by markuskowa** [comment]
@greydot I will merge it as soon as the changes mentioned above are made.

**Comment by betaboon** [comment]
@markuskowa i just adapted your requested changes
@greydot i also added the libbladeRF dependency.

I can say that it still compiles and works with my limesdr-mini (tested in gnuradio-companion and gqrx)

**Comment by markuskowa** [comment]
@GrahamcOfBorg build gnuradio-with-packages
@GrahamcOfBorg build gqrx qradiolink

**Comment by greydot** [comment]
I've just built and tested gqrx with bladeRF from this branch. Seems to work just fine.

**Comment by markuskowa** [comment]
Thanks for all effort!

**Comment by betaboon** [comment]
you're welcome. thanks for the nice and quick interaction to get this done :)

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 53646 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by jonringer** [comment]
hmm, awkward, mistyped my review script. Sorry

**Pull request merged**
